src/vm/reflectioninvocation.cpp

```diff
@@ -1383,8 +1383,18 @@ FCIMPL4(Object*, RuntimeMethodHandle::InvokeMethod,
         }
 #endif
 
+#ifdef _TARGET_ARM64_
+        ArgLocDesc argLocDescForStructInRegs;
```

Comment:
I still wonder why we have to differ for ARM64 here. Based on the IsHFA function above, the ArgDestination should have the m_argLocDescForStructInRegs set to NULL for non-HFA args. So if you used the original code at this place and just made sure that m_hasArgLocDescForStructInRegs in the ArgIterator is set only for HFA args, it seems it should work.